web/components/input/buttons/PrimaryButton/index.js

```diff
@@ -1,12 +1,13 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-const PrimaryButton = ({ text, onClick, disabled, className }) => (
+const PrimaryButton = ({ text, onClick, disabled, className, name, type }) => (
   <button
     onClick={onClick}
-    type="submit"
+    type={type ? type : (name ? "submit" : "button")}
     disabled={disabled}
-    className={`primary ${className}`}>
+    className={`primary ${className}`}
+    form={name}>
```

Comment:
do we really need ` form` here? It's not submitting even when the type is `submit` if `form` is defined

Reply:
Oops, `form` was supposed to equal the form-id but I made it the form-name. Fixed by fd5ad1c75187ec52ec1fd7a56af88ebb0f7eccdf. The commit also removes this attribute altogether since it's only really useful in the extreme case where you want to associate a button with a form when it's outside the `<form>` tags.